### PR TITLE
[BUGFIX] Changed the QueryBuilder for the Delete,Select and Update to avoid conflicts

### PR DESCRIPTION
**Pull request by tomasnorre**
This is not done yet, as we still cannot add Processes from backend.

But part of the problem is solved, there CommandController isn't throwing exceptions anymore.

### CODE REVIEW

File: Classes/Controller/CrawlerController.php [hunk 1]
```diff
@@ -2277,12 +2275,14 @@ public function CLI_run($countInARun, $sleepTime, $sleepAfterFinish)
         if (intval($this->extensionSettings['purgeQueueDays']) > 0) {
             $purgeDate = $this->getCurrentTime() - 24 * 60 * 60 * intval($this->extensionSettings['purgeQueueDays']);
 
-            $del = $queryBuilder
+            $queryBuilderDelete = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable($this->tableName);
```
Comment (tstahn) [inline Classes/Controller/CrawlerController.php]: Certainly nothing for now, but at some point we might want to split these database operations into separate methods. All this `$querBuilder*` juggling with its selections, deletions, updates, ... is pretty confusing.

Comment (tomasnorre) [reply Classes/Controller/CrawlerController.php]: Totally agree. This should be functions called instead, to keep it cleaner.

**Comment by bmack** [comment]
I think we should get it in as it is, however, there are places where we could use the Connection directly (e.g. UPDATE) instead of using another QueryBuilder instance.

**Comment by tomasnorre** [comment]
Will rebase and merge

File: Classes/Controller/CrawlerController.php [hunk 2]
```diff
@@ -2239,27 +2238,31 @@ public function CLI_run($countInARun, $sleepTime, $sleepAfterFinish)
...
+            $del = $queryBuilderDelete
                 ->delete($this->tableName)
                 ->where(
                     'exec_time != 0 AND exec_time < ' . $purgeDate
-                );
-            if (false == $del) {
-                $this->logger->info(
+                )->execute();
+
+            if (false === $del) {
```
Comment (tomasnorre) [inline Classes/Controller/CrawlerController.php]: Did I overwrite you log change @bmack ?

Comment (bmack) [reply Classes/Controller/CrawlerController.php]: looks ok

Comment (bmack) [reply Classes/Controller/CrawlerController.php]: oh yes, you did. instead of `getLogger()->log` use `logger->info(`

**Pull request merged**